Use Function Selector in Rule Builder

alexpaxton opened the pull request:
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass

Connect #1638 
Connect #1618 

### The Problem
- In the rule builder query maker applying aggregator functions to a field uses a dropdown, which is strangely broken
  - The component is rendering markup in the DOM but for some odd reason it is not appearing visually
  - Could not figure out why it was appearing incorrectly

### The Solution
- Use the `FunctionSelector` component instead of a dropdown (more consistent with the `QueryMaker` elsewhere)
- Modify `FunctionSelector` to optionally have a single selection mode

### Note
- I have done some testing on this, produces no errors or otherwise noticeable breakage
- Can't tell if this messed up the previous UX with function selection in the DE / CEO
- Not sure if this change will conflict with the auto apply function PR

121watts commented on ui/src/shared/components/FunctionSelector.js:
```diff
@@ -48,47 +54,65 @@ class FunctionSelector extends Component {
+                    {f}
+                  </div>
+                )
+              })}
```
It looks like these two maps do the same thing, but all they need are different functions passed to the onClick event handler.

alexpaxton commented:
@121watts combined the map methods and moved the ternary operator into the `onClick`
Also, added a bit of logic for users to deselect a function in the rule builder
